2021 07 todata dg eobj

agenius-logesh opened the pull request.

saranya-ag commented on R/cdfPlot.R:
```diff
@@ -83,15 +87,21 @@ cdfPlot <- function(contrastDF,
+    assertthat::assert_that(!missing(DGEdata),
+                            !is.null(DGEdata),
+                            "DGEobj" %in% class(DGEdata),
+                            msg = "DGEdata must be specified as class of DGEobj.")
```
Update to : DGEdata must be specified and must belong to DGEobj class.

agenius-logesh replied:
done for all

saranya-ag commented on R/cdfPlot.R:
```diff
@@ -83,15 +87,21 @@ cdfPlot <- function(contrastDF,
+
+    assertthat::assert_that(!missing(contrast),
+                            !is.null(contrast),
+                            contrast %in% names(DGEobj::getType(DGEdata, type = "topTable")),
```
Check for length of contrast

agenius-logesh replied:
done for all

saranya-ag commented on R/cdfPlot.R:
```diff
@@ -83,15 +87,21 @@ cdfPlot <- function(contrastDF,
+    assertthat::assert_that(!missing(contrast),
+                            !is.null(contrast),
+                            contrast %in% names(DGEobj::getType(DGEdata, type = "topTable")),
+                            msg = "contrast to be a singular value of class character and must be one from DGEdata with LogIntensity and LogRatio columns and optionally a p-value.")
```
Update to : contrast must be a singular value of class character and must be one of the top tables in the DGEdata

agenius-logesh replied:
done for all

saranya-ag commented on R/cdfPlot.R:
```diff
@@ -26,7 +26,9 @@
 #'
-#' @param contrastDF A dataframe with LogRatio and LogIntensity columns and optionally a p-value or FDR column.
+#' @param DGEdata Name of DGEobj with a class of DGEobj.
+#' @param contrast A character vector of a topTable data in DGEobj and its a class of dataframe
```
Name of a topTable item in DGEobj with LogRatio and LogIntensity columns and optionally a p-value or FDR column.

agenius-logesh replied:
done

saranya-ag commented on R/comparePlot.R:
```diff
@@ -29,8 +29,11 @@
-#' @param compareDF A dataframe with the first two columns representing the x and y variables.
-#'          Optionally add xp and yp columns to hold p-values or FDR values.
+#' @param DGEdata Name of DGEobj with a class of DGEobj.
+#' @param contrastOne and
```
combine this into one element - Accept a vector of two values.

agenius-logesh replied:
done

saranya-ag commented on R/comparePlot.R:
```diff
@@ -88,7 +96,10 @@
 #' @export
-comparePlot <- function(compareDF,
+comparePlot <- function(DGEdata,
+                        contrastOne,
```
Combine this into one vector.

agenius-logesh replied:
done

saranya-ag commented on R/plotDispersion.R:
```diff
@@ -69,89 +63,83 @@ plotDispersion <- function(DGEdata,
-            warning("linefitColor must be a singular value of class 'character' and must specify the name of the color or the rgb value. Assigning default value 'red'.")
-            linefitColor <- "red"
-        }
+    plotType <- tolower(plotType)
```
DGEdata is still a DGElist or counts matrix. you have not converted to DGEobj.

agenius-logesh replied:
done

saranya-ag commented on R/plotNorm.R:
```diff
@@ -38,20 +38,38 @@ plotNorm <- function(DGEdata,
                      plotType = "canvasXpress",
```
plotNorm - Is still accepting a counts matrix or DGEobj. This needs to be accept only DGEobj.

agenius-logesh replied:
done

saranya-ag commented on tests/testthat/test_cdfPlot.R:
```diff
@@ -2,34 +2,35 @@ context("DGEobj.plots - tests for cdfPlot.R functions")
+    contrast = "BDL_vs_Sham"
     # testing plot with default values.
-    plot <- cdfPlot(top_table, referenceLine = "blue")
+    plot <- cdfPlot(DGEdata, contrast, referenceLine = "blue")
```
For all test cases , specify both the parameter name and the value . You are specifying only the value for the first couple of parameters.

saranya-ag commented on tests/testthat/test_plotNorm.R:
```diff
@@ -82,10 +82,10 @@ test_that("plotNorm.R: plotNorm()", {
     expect_error(plotNorm(NULL),
                  regexp = "DGEdata must be of either class 'matrix' or 'DGEobj'.")
-    expect_error(plotNorm(t_obj1, plotCategory = "heatmap"),
+    expect_warning(plotNorm(t_obj1, plotCategory = "heatmap"),
```
check for the output class as well ..

saranya-ag replied:
Like the other warning statements ..

agenius-logesh replied:
done

agenius-logesh committed:
cdfplots changes done.

agenius-logesh committed:
comparePolt changes done.

agenius-logesh committed:
plotNorm changed to DGEobj.

agenius-logesh committed:
plotPValHist changes done.

agenius-mohammed-ali commented on R/cdfPlot.R:
```diff
@@ -26,7 +26,8 @@
 #' this order: Significant, Not Significant.
 #'
-#' @param contrastDF A dataframe with LogRatio and LogIntensity columns and optionally a p-value or FDR column.
+#' @param DGEdata Name of DGEobj with a class of DGEobj.
```
I do not think that is an accurate documentation. As user do not pass the object **name** she/he pass the object itself.

agenius-logesh replied:
done

agenius-mohammed-ali commented on R/cdfPlot.R:
```diff
@@ -54,16 +55,18 @@
-#'    cdfPlot(contrastDF, title = "My CDF Plot")
+#'    # Plot to console (DGEdata is a name of DGEonj and
+#'    contrast is a dataframe from DGEobj)
+#'    cdfPlot(DGEdata, contrast, title = "My CDF Plot")
```
- Please clarify in the example what value the **contrast** parameter can hold, so the example can be rerunnable.
-  Add another example for ggplot input

agenius-logesh replied:
done

agenius-logesh replied:
done

agenius-mohammed-ali commented on R/cdfPlot.R:
```diff
@@ -83,15 +86,22 @@ cdfPlot <- function(contrastDF,
     assertthat::assert_that(!is.null(pvalCol),
                             pvalCol %in% colnames(contrastDF),
-                            msg = "pvalCol column not found in contrastDF.")
+                            msg = "pvalCol column not found in contrast data.")
```
I think we need to check for the type of contrastDF here as well as the number of rows as old asserts.
(cc. @saranya-ag )

agenius-logesh replied:
I added the check for length.

agenius-mohammed-ali commented on R/comparePlot.R:
```diff
@@ -29,8 +29,10 @@
 #'
-#' @param compareDF A dataframe with the first two columns representing the x and y variables.
-#'          Optionally add xp and yp columns to hold p-values or FDR values.
+#' @param DGEdata Name of DGEobj with a class of DGEobj.
```
I do not think that is an accurate documentation. As user do not pass the object name she/he pass the object itself.

agenius-logesh replied:
done

agenius-mohammed-ali commented on R/comparePlot.R:
```diff
@@ -58,7 +60,10 @@
 #' \dontrun{
 #'   # Retrieve the first two contrasts from a DGEobj as a list of dataframes (length = 2; named items)
-#'   contrastList <- getType(DGEobj, "topTable")[1:2]
+#'   contrastList <- lapply(c(contrastOne,contrastTwo), function(x){
```
- Please provide the definition of `contrastOne, contrastTwo` parameters so user can understand how to create them and the example be runnable.
- Please add a ggplot example as well.

agenius-logesh replied:
done

agenius-mohammed-ali commented on R/comparePlot.R:
```diff
@@ -67,11 +72,12 @@
 #'
 #'   # Draw the plot
-#'   cPlot <- comparePlot(compareDat, title = "Plot Title")
+#'   cPlot <- comparePlot(DGEdata, contrasts, title = "Plot Title")
```
- How `contrasts` parameter is initialized, can we clarify that in the example?
- What its relation with the above preparation?

agenius-logesh replied:
done

agenius-mohammed-ali commented on R/comparePlot.R:
```diff
@@ -105,14 +113,43 @@ comparePlot <- function(compareDF,
+                            length(contrasts) == 2,
+                            contrasts[1] %in% names(DGEobj::getType(DGEdata, type = "topTable")),
+                            contrasts[2] %in% names(DGEobj::getType(DGEdata, type = "topTable")),
+                            msg = "contrast must be a class of character and must be two of the top tables in the DGEdata. with logFC and P.value columns.")
```
I think we need to check that the two columns are of type numeric as well

agenius-mohammed-ali replied:
@agenius-logesh - you missed this one

agenius-mohammed-ali commented on R/comparePlot.R:
```diff
@@ -105,14 +113,43 @@ comparePlot <- function(compareDF,
     assertthat::assert_that(plotType %in% c("canvasxpress", "ggplot"),
                             msg = "Plot type must be either canvasXpress or ggplot.")
+
+
```
extra unneeded space

agenius-logesh replied:
done

agenius-mohammed-ali commented on R/comparePlot.R:
```diff
@@ -262,6 +297,7 @@ comparePlot <- function(compareDF,
         dplyr::select(ylabel) %>%
         dplyr::summarise(across(everything(), list(min, max))) %>%
```
- Please add `accross` and `everything` functions to the import above
- Also call `dplyr::` before them

agenius-logesh replied:
done

agenius-mohammed-ali commented on R/comparePlot.R:
```diff
@@ -328,7 +364,7 @@ comparePlot <- function(compareDF,
                           symbolFill  = symbolColor)
         # Used to set uniform square scale
-        scalemax = compareDF[,1:2] %>% as.matrix %>% abs %>% max %>% multiply_by(1.05)
+        scalemax = compareDF[,1:2] %>% as.matrix %>% abs %>% max %>% magrittr::multiply_by(1.05)
```
- Please add `multiply_by` to the importFrom statements above

agenius-logesh replied:
done

agenius-mohammed-ali commented on R/internal.R:
```diff
@@ -40,13 +40,17 @@
 
 .get_valid_symbolShapes_cxplot <- function() {
-    valid_shapes <- c("sphere", "square", "rhombus", "triangle", "plus", "star", "octagon", "oval",
+    valid_shapes <- c("circle","sphere", "square", "rhombus", "triangle", "plus", "star", "octagon", "oval",
```
- Defining `valid_shapes` is useless here, you can return the vector directly

agenius-mohammed-ali replied:
@agenius-logesh - you missed this one

agenius-mohammed-ali commented on R/obsPlot.R:
```diff
@@ -1,274 +1,466 @@
-#'   Allowed values: "fixed", "free_x", "free_y", "free")
+#'   (FALSE)  (default = TRUE)
+#' @param axisFree Specify same scale or independent scales for each subplot (default = TRUE;
+#'   Allowed values: TRUE and FALSE)
```
- Rephrase to: **Allowed values: TRUE or FALSE**

agenius-logesh replied:
done

agenius-mohammed-ali commented on R/obsPlot.R:
```diff
@@ -1,274 +1,466 @@
+        boxColor <- NA
+    } else {
+        violinColor <- NA
+        boxColor <- color
```
- cannot we use one variable for color instead of two regardless the layer?

agenius-logesh replied:
done

agenius-mohammed-ali commented on R/obsPlot.R:
```diff
@@ -1,274 +1,466 @@
-            # Rotate xaxis group labels
-            if (xAngle > 0) {
-                aplot <- aplot + theme(axis.text.x = element_text(angle = xAngle, hjust = 1))
+        .addGeoms <- function(obsPlot) {
```
- Call `ggplot::` before ggplot related functions

agenius-logesh replied:
done

agenius-mohammed-ali commented on R/plotNorm.R:
```diff
@@ -8,7 +8,7 @@
 #' data.  Thus this function is only appropriate for count-based data.
 #'
-#' @param DGEdata  A DGEobj or counts matrix.
+#' @param DGEdata Name of DGEobj with a class of DGEobj.
```
I do not think that is an accurate documentation. As user do not pass the object name she/he pass the object itself.

agenius-logesh replied:
done

agenius-mohammed-ali commented on R/plotPValHist.R:
```diff
@@ -4,7 +4,8 @@
 #' but should be useful for any dataframe of numeric columns.
 #'
-#' @param P.Val A matrix or dataframe of numeric data; col = samples
+#' @param DGEdata Name of DGEobj with a class of DGEobj.
```
I do not think that is an accurate documentation. As user do not pass the object name she/he pass the object itself.

agenius-logesh replied:
done

agenius-mohammed-ali commented on R/plotPValHist.R:
```diff
@@ -16,63 +17,78 @@
 
-    assertthat::assert_that(all(sapply(data.frame(P.Val),is.numeric)),
-                            msg = "P.Val must contain only numeric values.")
+    P.Val <- extractCol(getType(DGEdata, "topTable"), colName = P.Val, robust = FALSE)
```
need to mention what packages function `getType` and `extractCol`  belongs to

agenius-logesh replied:
done

agenius-mohammed-ali commented on R/plotPValHist.R:
```diff
@@ -16,63 +17,78 @@
+            length(facet) != 1)) {
         warning("facet must be a singular logical value. Assigning default value TRUE.")
-        facet <- TRUE
+        facet = TRUE
```
Why using "=" sign here instead of the usual `<-`?

agenius-logesh replied:
changed.

agenius-mohammed-ali reviewed (changes requested):
Do we need a separate unit tests file for internals.R?
(cc. @saranya-ag )

agenius-logesh committed:
comparePlot changes done.

agenius-logesh committed:
obsPlot plotNorm and plotpValHist changes done.

agenius-logesh committed:
cdfPlots changes done.

agenius-logesh committed:
cdfplot final update.

agenius-mohammed-ali reviewed (commented):
There are two comments only remaining.

agenius-logesh committed:
comparePlot changes done.

agenius-logesh committed:
plotPValHist changes done.

agenius-logesh committed:
obsPlot changes done.

agenius-logesh committed:
comparePlot assert condition modified.

agenius-logesh committed:
cdfPlot changes done.

agenius-logesh committed:
plotDispersion changed to DGEobj.

agenius-logesh committed:
plotDispersion chnaged.

agenius-logesh committed:
DGEdata to dgeObj changed.
ggplot2:: removed.
plotType, contrast, xlab ylab Titile matched.

agenius-logesh committed:
conflict resolve.

Merge remote-tracking branch 'origin/2021-07-refactor-all-plots' into 2021-07-todata-DGEobj

# Conflicts:
#	R/cdfPlot.R
#	R/comparePlot.R
#	R/plotDispersion.R
#	tests/testthat/test_cdfPlot.R
#	tests/testthat/test_comparePlot.R
#	tests/testthat/test_ggplotMDS.R
#	tests/testthat/test_plotDispersion.R
#	tests/testthat/test_plotPValHist.R
